rustls-post-quantum/src/mlkem.rs

```diff
@@ -0,0 +1,96 @@
+use aws_lc_rs::kem;
+use aws_lc_rs::unstable::kem::ML_KEM_768;
+use rustls::crypto::{ActiveKeyExchange, CompletedKeyExchange, SharedSecret, SupportedKxGroup};
+use rustls::ffdhe_groups::FfdheGroup;
+use rustls::{Error, NamedGroup, PeerMisbehaved, ProtocolVersion};
+
+/// This is the [MLKEM] key exchange.
+///
+/// [MLKEM]: https://datatracker.ietf.org/doc/draft-connolly-tls-mlkem-key-agreement
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug)]
+pub struct MLKEM768;
+
+impl SupportedKxGroup for MLKEM768 {
+    fn start(&self) -> Result<Box<dyn ActiveKeyExchange>, Error> {
+        let decaps_key = kem::DecapsulationKey::generate(&ML_KEM_768)
+            .map_err(|_| Error::FailedToGetRandomBytes)?;
```

Comment:
both `kem::DecapsulationKey::generate()` and `DecapsulationKey::encapsulation_key()` are doc'd as returning an error "when operation fails due to internal error."

Is `FailedToGetRandomBytes` the best choice for these error conditions?